util/openapi-generator.ts

```diff
@@ -0,0 +1,31 @@
+import Fastify, { FastifyPluginAsync } from 'fastify';
+import { TypeBoxTypeProvider } from '@fastify/type-provider-typebox';
+import { Api, ApiSwaggerOptions } from '../src/api/init';
+import FastifySwagger from '@fastify/swagger';
+import { existsSync, mkdirSync, writeFileSync } from 'fs';
+import { Server } from 'http';
+
+/**
+ * Generates `openapi.yaml` based on current Swagger definitions.
+ */
+export const ApiGenerator: FastifyPluginAsync<
+  Record<never, never>,
+  Server,
+  TypeBoxTypeProvider
+> = async (fastify, options) => {
+  await fastify.register(FastifySwagger, ApiSwaggerOptions);
+  await fastify.register(Api);
+  if (!existsSync('./tmp')) {
+    mkdirSync('./tmp');
+  }
+  writeFileSync('./tmp/openapi.yaml', fastify.swagger({ yaml: true }));
```

Comment:
(Possibly for a later PR) looks like a client library could easily be generated as well, since this already outputs the flattened schema file, any thoughts?

Reply:
Agreed 👍  Opened https://github.com/hirosystems/token-metadata-service/issues/65